great_expectations/checkpoint/checkpoint.py

```diff
@@ -804,7 +807,8 @@ def __init__(
             template_name=template_name,
             run_name_template=run_name_template,
             expectation_suite_name=expectation_suite_name,
-            batch_request=batch_request,
+            batch_request=batch_request,  # type: ignore[arg-type] # FluentBatchRequest is not a dict
+            # TODO: check if `pydantic.BaseModel` and call `batch_request.dict()`??
```

Comment:
Wanted to call this out because I would expect for things to break here if we passed a `FluentBatchRequest` here as is.

Reply:
hmm good call - could maybe be a good ticket for an FDS cleanup epic?

Reply:
The Config objects that extend `DictDot` only accept primitive types -- by design; hence `batch_request` is always a `dict`, meaning that the `ignore` statement may not be appropriate.  However, I would like to point out that this `Marshmallow`-based design is overdue for being implemented on `Pydantic` instead, where typed objects will be the norm.  Thanks.

Reply:
The ignore IS needed, `batch_request` is NOT always a dict.

Reply:
@Kilo59 This sounds very concerning.  I am looking for a test where we submit a non-`dict` entity to Config.

Reply:
@Kilo59 Could you please take a look at line 2810 in `great_expectations/data_context/types/base.py` -- do you see why I am concerned?  Thank you.

Reply:
It is unclear why `CheckpointConfig` in that module allows only plain Python `dict` as `batch_request` while yet the `ignore` seems to be needed.  We will need to understand what is happening there in detail later.

Reply:
@alexsherstinsky yes, this looks like it's currently (and has been broken).